src/zarr/storage/__init__.py

```diff
@@ -17,3 +22,20 @@
     "ZipStore",
     "make_store_path",
 ]
+
+
+class VerboseModule(ModuleType):
+    def __setattr__(self, attr: str, value: Any) -> None:
+        if attr == "default_compressor":
+            warnings.warn(
+                "setting zarr.storage.default_compressor is deprecated, use "
+                "zarr.config to configure array.v2_default_compressor "
+                "e.g. config.set({'codecs.zstd':'your.module.Zstd', 'array.v2_default_compressor.numeric': 'zstd'})",
+                DeprecationWarning,
+                stacklevel=1,
```

Comment:
not sure if this is the correct stacklevel

Reply:
I think this is fine.